java/openjdk-java/portfile

```diff
@@ -0,0 +1,380 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem          1.0
+name                openjdk-java
+version             17
+categories          java
+platforms           darwin
+description         an Open Source, Open Innovation, Open Standards implementation of the Java Platform, Standard Edition, and related projects.
+homepage            openjdk.java.net
+license             GPL-2+CE
+universal_variant   no
+
+set long_description_openjdk-ga \
+ Production-ready open-source builds of the Java Development Kit
+
+set long_description_openjdk-ea \
+ early-access, open-source builds
+
+set long_description_openjfx \
+ JavaFX is an open source, next generation client application platform for desktop, mobile and embedded systems built on Java. It is a collaborative effort by many individuals and companies with the goal of producing a modern, efficient, and fully featured toolkit for developing rich client applications.
+
+set obsoleted_ports {
+openjdk-java
+}
+
+if {${subport} in ${obsoleted_ports}} {
+    PortGroup    obsolete 1.0
+
+}
+
+if {${subport} eq "openjdk-java"} {
+    # The openjdk-java is not installable, but recommends users to install the latest java subport
+    replaced_by  openjdk${version}
+}
+
+subport openjdk-6-bsd-i386-soylatte {
+ version             1.6
+ set build           1.0.3
+ description         openjdk
+ supported archs     i386
+ long_description    openjdk 6
+ master_sites        https://landonf.org/static/soylatte/bsd-dist/javasrc_${version}_jrl_darwin/soylatte16-i386-${build}.tar.bz2
+ distname            soylatte16-i386-${build}
+ worksrcdir          soylatte16-i386-${build}
+}
+
+subport openjdk-7-bsd-ppc-zerovm {
+ version             1.7
+ Set build           2012-02-16-07-18-b00
+ description         openjdk 7 port:bsd builds
+ supported_archs     ppc 
+ long_description    openjdk-7 darwin port:bsd ppc builds
+ master_sites        http://commondatastorage.googleapis.com/bluelotussoftware/openjdk/macppc/openjdk7-zerovm-ppc-jyeary-${build}.zip
+ distname            openjdk7-zerovm-ppc-jyeary-${build}
+ worksrcdir          openjdk7-zerovm-ppc-jyeary-${build}
+}
+
+subport openjdk-9-jdk {
+ version             9.0.4
+ description         openjdk 9 builds
+ long_description    ${long_description_openjdk-ga}
+ master_sites        https://download.java.net/java/GA/jdk9/${version}/binaries/openjdk-{version}_osx-x64_bin.tar.gz
+ supported_archs     x86_64
+ checksum            sha256  66415406716fc42cff36e2d74ae991d46c42b3cf317b0425c7bf67697c616716
```

Comment:
This is a syntax error. It's `checksums` not `checksum`.

Please list three checksum types for each distfile: rmd160, sha256, and size.

Many of the subports seem to be missing checksums entirely.

Reply:
Actually there is no rmd160 and size (in bytes)

Reply:
openjdk-6-bsd-i386-soylatte and openjdk-7-bsd-ppc-zerovm are missing checksum

Reply:
fixed

Reply:
fixed syntax error and added rmd160, sha256 and size to subports